app/controllers/internal/configs_controller.rb

```diff
@@ -31,6 +32,11 @@ def config_params
     params.require(:site_config).permit(allowed_params)
   end
 
+  def confirmation
+    not_authorized unless current_user.has_role?(:single_resource_admin, Config) # Special additional permission
```

Comment:
I added this here instead of via pundit because this seemed like an explicit additional step here, and colocated it with the other confirmation check. Open to suggestions if this is not ideal.

Reply:
any reason to not make a it a `before_action` for the `create` action here and put inline?

Reply:
Hmmm. I had put it here to be more explicit, but since we do this with other kinds of authorization that does make sense, I'll make this change.